src/vs/workbench/contrib/webview/browser/webview.ts

```diff
@@ -329,7 +329,28 @@ export class WebviewOriginStore {
 		return newOrigin;
 	}
 
-	private getKey(viewType: string, extId: ExtensionIdentifier): string {
-		return JSON.stringify({ viewType, extension: extId.value });
+	private getKey(viewType: string, additionalKey: string | undefined): string {
+		return JSON.stringify({ viewType, key: additionalKey });
+	}
+}
+
+/**
+ * Stores the unique origins for a webview.
+ *
+ * These are randomly generated, but keyed on extension and webview viewType.
+ */
+export class ExtensionKeyedWebviewOriginStore {
+
+	private readonly store: WebviewOriginStore;
+
+	constructor(
+		rootStorageKey: string,
+		@IStorageService storageService: IStorageService,
```

Comment:
@mjbvz For the web worker extension host we actually use `localStorage` (https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) for this. This brings the benefit of being able to invoke "clear website data" browser commands for the actual page and it drops the data for the iframes as well

Reply:
oh, you are right. I was mistaken. Maybe the storage service on browsers is actually backed by localstorage. Thanks for following up